certora/specs/moreExamplesERC20.spec

```diff
@@ -0,0 +1,71 @@
+/***
+
+Spec for the Certora Verification Language. 
+See https://docs.certora.com for a complete guide.
+
+run this file with ghostWithERc20s.conf:
+
+certoraRun certora\confs\ghostWithERc20s.conf 
+
+***/
+
+using ERC20Helper as _erc20Helper; // a variable that is the address of ERC20Helper  
+
+using SimpleERC20 as simpleERC20;
+
+// Declaration of erc20 functions 
+methods {
+    // When there is an unresolved call it will check all implementaions found in the list of contracts 
+    // See docs for more information on dispatcher summary:
+    // https://docs.certora.com/en/latest/docs/cvl/methods.html?highlight=dispatcher#dispatcher-summaries
+    function _.name() external => DISPATCHER(true);
+    function _.symbol() external => DISPATCHER(true);
+    function _.decimals() external => DISPATCHER(true);
+    function _.totalSupply() external => DISPATCHER(true);
+    function _.balanceOf(address) external => DISPATCHER(true);
+    function _.allowance(address,address) external => DISPATCHER(true);
+    function _.approve(address,uint256) external => DISPATCHER(true);
+    function _.transfer(address,uint256) external => DISPATCHER(true);
+    function _.transferFrom(address,address,uint256) external => DISPATCHER(true);
+    
+    // functions in other contracts
+	  function ERC20Helper.tokenBalanceOf(address token, address user) external returns (uint256) envfree;
+
+    function simpleERC20.totalSupply() external returns (uint256) envfree;
+}
+
+
+/**
+    @title decrease in system's erc20 balance
+    @dev calls another contract 
+    @dev example for filtering and checking only on methods from the main contract 
+    @notice method f must be defined in the rule signature to be recognized by the filter
+*/ 
+rule decreaseInERC20(method f, address token) filtered { f-> f.contract == currentContract } {
+   
+    uint256 before = _erc20Helper.tokenBalanceOf(token, currentContract);
+
+    env e;
+	  calldataarg arg;
+    f(e, arg);
+
+    uint256 after = _erc20Helper.tokenBalanceOf(token, currentContract);
+
+    assert after >= before ||  false ; //fill in cases token can decrease 
+
+} 
+
+/**
+    @title sum of erc20 balances is totalSupply 
+    @dev usign a ghost variable and hooks to track changes to _balances
```

Comment:
```suggestion
    @dev using a ghost variable and hooks to track changes to _balances
```

Reply:
Link to ghosts?
https://docs.certora.com/en/latest/docs/confluence/anatomy/ghostfunctions.html